utils/app/const.ts

```diff
@@ -43,7 +43,7 @@ export const OPENAI_API_HOST_TYPE =
     process.env.OPEN_AI_HOST_TYPE || 'apim';
 
 const COMMON_CONFIGURATION: any = {
-    OPENAI_API_VERSION: "2023-03-15-preview",
+    OPENAI_API_VERSION: OPENAI_API_VERSION,
```

Comment:
This is really the only part that fixes the issue of the version being hard-coded. Everything else is just setting up appropriate defaults